packages/dds/tree/src/feature-libraries/defaultFieldKinds.ts

```diff
@@ -532,29 +605,63 @@ const optionalChangeRebaser: FieldChangeRebaser<OptionalChangeset> = isolatedFie
 			if (over.fieldChange !== undefined) {
 				const wasEmpty = over.fieldChange.newContent === undefined;
 
-				// We don't have to rebase the child changes, since the other child changes don't apply to the same node
+				// TODO: Handle rebasing child changes.
 				return {
 					...change,
 					fieldChange: { ...change.fieldChange, wasEmpty },
 				};
 			}
 
-			return change;
+			const rebasedChange = { ...change };
+			const overChildChange =
+				change.deletedBy === over.deletedBy ? over.childChange : undefined;
+			const rebasedChildChange = rebaseChild(change.childChange, overChildChange);
+			if (rebasedChildChange !== undefined) {
+				rebasedChange.childChange = rebasedChildChange;
+			} else {
+				delete rebasedChange.childChange;
+			}
+
+			return rebasedChange;
 		}
 
 		if (change.childChange !== undefined) {
 			if (over.fieldChange !== undefined) {
-				// The node the child changes applied to no longer exists so we drop the changes.
-				// TODO: Represent muted changes
-				return {};
+				if (change.deletedBy === undefined) {
+					// `change.childChange` refers to the node being deleted by `over`.
+					return {
+						childChange: rebaseChild(change.childChange, over.childChange),
+						deletedBy: overTagged.revision,
+					};
+				} else if (
+					over.fieldChange.newContent !== undefined &&
+					"revert" in over.fieldChange.newContent &&
+					over.fieldChange.newContent.revision === change.deletedBy
+				) {
+					// Remove deletedBy because we revived the node that childChange refers to
+					return {
+						childChange: rebaseChild(
+							change.childChange,
+							over.fieldChange.newContent.changes,
+						),
+					};
+				}
 			}
+		}
 
-			if (over.childChange !== undefined) {
-				return { childChange: rebaseChild(change.childChange, over.childChange) };
+		{
+			const rebasedChange = { ...change };
+			const overChildChange =
+				change.deletedBy === over.deletedBy ? over.childChange : undefined;
+			const rebasedChildChange = rebaseChild(change.childChange, overChildChange);
+			if (rebasedChildChange !== undefined) {
+				rebasedChange.childChange = rebasedChildChange;
+			} else {
+				delete rebasedChange.childChange;
 			}
-		}
 
-		return change;
+			return rebasedChange;
+		}
```

Comment:
I'm a little confused by the presence of this block of code: it seems like we should only reach this area if  we have an empty changeset, which doesn't really need rebasing.

Reply:
We should probably reorganize the code to make this clearer, but this block will be reached if `change.fieldChange === undefined` and `over.fieldChange` (if any) does not affect `change.childChange`. In particular, this block handles the case where `change` and `over` both only have child changes. `over` may also be empty, but the new rebase contract requires rebasing child changes even when there is no corresponding base change.

Reply:
Ha, thanks, I missed the fact that the if block above doesn't always return. 
Based on that, it seems this code can call `rebaseChild(change.childChange, overChildChange)` when both changesets have child changes and when `change.deletedBy` is not undefined. This is a problem because when that's the case, the child changes of the repsective changesets are meant to affect different nodes.